src/config.rs

```diff
@@ -690,3 +690,457 @@ impl TiKvConfig {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use {raftstore, server, storage};
+    use util::config::compression_type_level_serde;
+
+    use toml::{self, ser};
+
+    extern crate serde_test;
+    use self::serde_test::{assert_de_tokens, Token};
+
+    #[test]
+    fn test_toml_serde_roundtrippping() {
+        {
+            let value = DbConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = DefaultCfConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = LockCfConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = MetricConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = PdConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = RaftCfConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = RaftDbConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = RaftDefaultCfConfig::default();
+            let dump = ser::to_string_pretty(&value).unwrap();
+            let load = toml::from_str(&dump).unwrap();
+            assert_eq!(value, load);
+        }
+        {
+            let value = TiKvConfig::default();
```

Comment:
I think this has covered all the other cases.

Reply:
Yes. I think it ok to unit test all of them, these tests are fast. And if the tests fails, we can find out which one breaks it.

Reply:
Actually if they are not equal, the text will be printed out.